pkg/build/builder/util.go

```diff
@@ -46,26 +56,83 @@ func readNetClsCGroup(reader io.Reader) string {
 	names := []string{"net_cls", "cpu"}
 	for _, group := range names {
 		if value, ok := cgroups[group]; ok {
-			return value
+			return value, containerType
+		}
+	}
+
+	return "", containerType
+}
+
+// readResolveConfHostPath determines the path to the resolv.conf file for this
+// container, as it exists on the host machine.  (/etc/resolv.conf is mounted
+// into the container from the host path).
+func readResolvConfHostPath() (string, error) {
```

Comment:
This stuff needs to go into its own package.  It's getting pretty ugly in here.  These aren't build utils, they're build environment / os coupling.

Reply:
do you really want to encourage other people to consume this stuff?  again it should all go away in the future.